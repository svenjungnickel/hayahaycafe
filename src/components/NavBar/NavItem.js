import React from 'react';
import PropTypes from 'prop-types';
import { Nav } from 'react-bootstrap';
import { Link } from 'gatsby';
import { navBar__item } from '../../styles/components/NavBar/NavBar.module.scss';

const NavItem = ({ link, name, active }) => (
    <Nav.Item className={navBar__item}>
        <Link
            to={link}
            className={true === active ? 'nav-link active' : 'nav-link'}
            data-cy={`navBarLink${name.replace(/\s/g, '')}`}
        >
            {name}
        </Link>
    </Nav.Item>
);

NavItem.defaultProps = {
    active: false,
};

NavItem.propTypes = {
    link: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    active: PropTypes.bool,
};

export default NavItem;
