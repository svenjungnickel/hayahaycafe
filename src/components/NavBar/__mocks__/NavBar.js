import React from 'react';

const NavBar = () => <nav />;

export default NavBar;
