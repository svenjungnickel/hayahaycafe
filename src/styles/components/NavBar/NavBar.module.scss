@import "../../mixins/media-queries";
@import "../../mixins/transition";

.navBar {
    background-color: var(--navBar-bg);
    text-transform: uppercase;
    color: var(--color-primary);
    letter-spacing: 2px;
    font-weight: 700;

    .navBar__items {
        display: flex;
        align-items: center;

        .navBar__item {
            a {
                color: var(--color-primary);

                &:focus,
                &:hover {
                    color: var(--color-secondary);
                }
            }
        }
    }

    .navBar__logo {
        transition: width $transition-duration $transition-timing-function, height $transition-duration $transition-timing-function;
        margin: 0 20px;
        width: 97px;
        height: 80px;

        @include respond-above(sm) {
            width: 146px;
            height: 120px;
        }
    }
}

.navBar-small {
    @extend .navBar;

    .navBar__logo {
        width: 48px;
        height: 40px;
    }
}
