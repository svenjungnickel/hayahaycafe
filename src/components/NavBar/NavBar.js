import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { StaticQuery, graphql } from 'gatsby';
import { Navbar, Nav } from 'react-bootstrap';
import useDarkMode from 'use-dark-mode';
import Logo from './Logo';
import NavItem from './NavItem';
import DarkModeToggle from './DarkModeToggle';
import NavBarStyles from '../../styles/components/NavBar/NavBar.module.scss';
import DarkModeToggleStyle from '../../styles/components/NavBar/DarkModeToggle.module.scss';

const NavBar = ({ currentPage }) => {
    currentPage = `/${currentPage}`;
    const darkMode = useDarkMode(false);
    const [smallNavBar, setSmallNavBar] = useState(false);

    useEffect(() => {
        window.addEventListener('scroll', () => {
            // minimize nav bar only when scrolled past the nav bar height to avoid annoying flickering
            setSmallNavBar(window.pageYOffset > 135);
        });
    });

    return (
        <StaticQuery
            query={graphql`
                query {
                    logoMobile: file(relativePath: { eq: "logo.png" }) {
                        childImageSharp {
                            fixed(height: 80) {
                                ...GatsbyImageSharpFixed
                            }
                        }
                    }
                    logo: file(relativePath: { eq: "logo.png" }) {
                        childImageSharp {
                            fixed(height: 120) {
                                ...GatsbyImageSharpFixed
                            }
                        }
                    }
                }
            `}
            render={(data) => (
                <Navbar
                    collapseOnSelect
                    variant={true === darkMode.value ? 'dark' : 'light'}
                    expand="md"
                    fixed="top"
                    id="site-navbar"
                    className={true === smallNavBar ? NavBarStyles.navBarSmall : NavBarStyles.navBar}
                    data-cy="navBar"
                >
                    <Logo logo={data.logoMobile.childImageSharp.fixed} className="d-block d-md-none" />
                    <DarkModeToggle darkMode={darkMode} className="d-block d-md-none" />
                    <Navbar.Toggle aria-controls="basic-navbar-nav" className="ml-right" data-cy="navBarToggle" />
                    <Navbar.Collapse id="basic-navbar-nav" className="justify-content-center">
                        <Nav className={NavBarStyles.navBar__items}>
                            <NavItem link="/" name="Home" active={currentPage === '/'} />
                            <NavItem link="/about" name="About" active={currentPage === '/about'} />
                            <Logo logo={data.logo.childImageSharp.fixed} className="d-none d-md-block" />
                            <NavItem link="/gallery" name="Gallery" active={currentPage === '/gallery'} />
                            <NavItem link="/contact" name="Contact" active={currentPage === '/contact'} />
                        </Nav>
                    </Navbar.Collapse>
                    <div className={DarkModeToggleStyle.darkModeToggle}>
                        <DarkModeToggle darkMode={darkMode} className="d-none d-md-block" />
                    </div>
                </Navbar>
            )}
        />
    );
};

NavBar.propTypes = {
    currentPage: PropTypes.string.isRequired,
};

export default NavBar;
