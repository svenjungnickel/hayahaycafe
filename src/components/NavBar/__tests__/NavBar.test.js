import React from 'react';
import renderer from 'react-test-renderer';
import { StaticQuery } from 'gatsby';
import NavBar from '../NavBar';

jest.mock('../Logo');
jest.mock('../NavItem');
jest.mock('../DarkModeToggle');

describe('NavBar', () => {
    it('renders without currentPage throws prop type error', () => {
        const renderNavBar = () => {
            const component = <NavBar />;
            renderer.create(component);
        };

        expect(renderNavBar).toThrowError('Warning: Failed prop type');
    });

    it('renders with invalid currentPage throws prop type error', () => {
        const renderNavBar = () => {
            const currentPage = 123;

            const component = <NavBar currentPage={currentPage} />;
            renderer.create(component);
        };

        expect(renderNavBar).toThrowError('Warning: Failed prop type');
    });

    it('renders with disabled dark mode (default)', () => {
        const currentPage = '/';
        const data = {
            logoMobile: {
                childImageSharp: {
                    fixed: {
                        src: 'src',
                    },
                },
            },
            logo: {
                childImageSharp: {
                    fixed: {
                        src: 'src',
                    },
                },
            },
        };
        StaticQuery.mockImplementationOnce(({ render }) => render(data));

        const component = <NavBar currentPage={currentPage} />;
        const tree = renderer.create(component).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
