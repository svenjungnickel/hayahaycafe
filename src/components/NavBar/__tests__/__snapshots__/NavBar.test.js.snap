// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar renders with disabled dark mode (default) 1`] = `
<nav
  className="navBar navbar navbar-expand-md navbar-light fixed-top"
  data-cy="navBar"
  id="site-navbar"
>
  <img
    alt="logo"
    src="logo"
  />
  <input
    role="switch"
    type="checkbox"
  />
  <button
    aria-controls="basic-navbar-nav"
    aria-label="Toggle navigation"
    className="ml-right navbar-toggler collapsed"
    data-cy="navBarToggle"
    onClick={[Function]}
    type="button"
  >
    <span
      className="navbar-toggler-icon"
    />
  </button>
  <div
    aria-expanded={null}
    className="justify-content-center navbar-collapse collapse"
    id="basic-navbar-nav"
  >
    <div
      className="navBar__items navbar-nav"
      onKeyDown={[Function]}
    >
      <nav />
      <img
        alt="logo"
        src="logo"
      />
      <nav />
    </div>
  </div>
  <div
    className="darkModeToggle"
  >
    <input
      role="switch"
      type="checkbox"
    />
  </div>
</nav>
`;
